File: VB/CheckCompatibility/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace CheckCompatibility
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Shared Function IsCompatible(ByVal chart As ChartControl, ByVal type As ViewType) As Boolean
			Dim test As New ChartControl()
			Dim series As New Series("testSeries", type)
			test.Series.Add(series)
			Return test.Diagram.GetType().Equals(chart.Diagram.GetType())
		End Function

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			If IsCompatible(Me.chartControl1, ViewType.Area) Then
				Me.chartControl1.Series(0).ChangeView(ViewType.Area)
			Else
				MessageBox.Show("Incompatible!")
			End If
		End Sub

		Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
			If IsCompatible(Me.chartControl1, ViewType.Line) Then
				Me.chartControl1.Series(0).ChangeView(ViewType.Line)
			Else
				MessageBox.Show("Incompatible!")
			End If
		End Sub

		Private Sub button3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button3.Click
			If IsCompatible(Me.chartControl1, ViewType.Pie) Then
				Me.chartControl1.Series(0).ChangeView(ViewType.Pie)
			Else
				MessageBox.Show("Incompatible!")
			End If
		End Sub
	End Class
End Namespace

File: CS/CheckCompatibility/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace CheckCompatibility
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        static bool IsCompatible(ChartControl chart, ViewType type)
        {
            ChartControl test = new ChartControl();
            Series series = new Series("testSeries", type);
            test.Series.Add(series);            
            return test.Diagram.GetType().Equals(chart.Diagram.GetType());
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if(IsCompatible(this.chartControl1, ViewType.Area))
                this.chartControl1.Series[0].ChangeView(ViewType.Area);
            else
                MessageBox.Show("Incompatible!");
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if(IsCompatible(this.chartControl1, ViewType.Line))
                this.chartControl1.Series[0].ChangeView(ViewType.Line);
            else
                MessageBox.Show("Incompatible!");
        }

        private void button3_Click(object sender, EventArgs e)
        {
            if(IsCompatible(this.chartControl1, ViewType.Pie))
                this.chartControl1.Series[0].ChangeView(ViewType.Pie);
            else
                MessageBox.Show("Incompatible!");
        }
    }
}